staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go

```diff
@@ -375,6 +376,33 @@ func NewServerTimeout(qualifiedResource schema.GroupResource, operation string,
 	}}
 }
 
+func NewStorageReadError(qualifiedResource schema.GroupResource, resourcePrefix string, keysToError map[string]error) *StatusError {
+	var causes []metav1.StatusCause
+	allKeys := make([]string, 0, len(keysToError))
+	for failedKey, err := range keysToError {
+		allKeys = append(allKeys, failedKey)
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeUnexpectedServerResponse,
+			Field:   failedKey,
+			Message: err.Error(),
```

Comment:
question: including the error message in the response, it might have a security concern?

Reply:
I'd like to think not but I'll try to have a look